src/components/pages/home/slider/MainSlider.js

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable max-len */
+import { bannersData } from "../../../../mocks/featured-banners-data";
+import { useEffect, useState } from "react";
+
+//Components
+import { BtnSlider } from "../../../common/BtnSlider";
+import { Slider } from "../../../common/Slide";
+
+const banners = bannersData.results;
+
+export function MainSlider() {
+  const [slideIndex, setSlideIndex] = useState(1);
+  const [paused, setPaused] = useState(false);
+
+  const nextSlide = () => {
+    if (slideIndex !== banners.length) {
+      setSlideIndex(slideIndex + 1);
+    } else if (slideIndex === banners.length) {
+      setSlideIndex(1);
+    }
+  };
+
+  const prevSlide = () => {
+    if (slideIndex !== 1) {
+      setSlideIndex(slideIndex - 1);
+    } else if (slideIndex === 1) {
+      setSlideIndex(banners.length);
+    }
+  };
+
+  const moveDot = (index) => {
+    setSlideIndex(index);
+  };
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if (!paused) {
+        if (slideIndex !== banners.length) {
+          setSlideIndex(slideIndex + 1);
+        } else if (slideIndex === banners.length) {
+          setSlideIndex(1);
+        }
```

Comment:
this is also a case:

```suggestion
        const newIndex = slideIndex !== banners.length
          ? slideIndex + 1
          : 1;
        
        setSlideIndex(newIndex);
```

Reply:
Solved